src/analysis/retail/evoker/preservation/normalizers/CastLinkNormalizer.ts

```diff
@@ -1062,4 +1074,19 @@ export function isEbFromReversion(
   return HasRelatedEvent(event, EB_REVERSION);
 }
 
+export function getTimeOfNeedHealing(event: SummonEvent) {
+  const allHealing = GetRelatedEvents<HealEvent>(event, TIME_OF_NEED_HEALING) ?? null;
+  const tonHealing: HealEvent[] = [];
+  if (allHealing) {
+    allHealing.forEach((heal) => {
+      //Only grab heals that come from the spawned entity.
+      //This is to prevent Player VE's from getting pulled cause they have the same id
+      if (heal.sourceID === event.targetID) {
+        tonHealing.push(heal);
+      }
+    });
```

Comment:
rather than doing this filtering here, you can do it during normalization by specifying an `additionalCondition` above like:

```tsx
additionalCondition: (linkingEvent, referencedEvent) => (linkingEvent as SummonEvent).targetID === (referencedEvent as HealEvent).sourceID
```

(the `as <subtype>` clauses are unfortunately needed because of the `AnyEvent` parameter types on the interface).